lib/base-compiler.ts

```diff
@@ -338,7 +368,7 @@ export class BaseCompiler implements ICompiler {
         return {mtime: this.mtime, compiler, args, options};
     }
 
-    protected async execCompilerCached(compiler, args, options) {
+    protected async execCompilerCached(compiler, args, options, errorsAreOk = false) {
```

Comment:
I'm pretty confused by this, there's a lot of double-negatives going on here. if (!errorsareOk)... etc etc.

I know you hit issues with this caching stuff, but I don't get quite what's going on. Looks like now the default is:

- errorsAreOk is false
- if the cached result has any stderr (even if it didn't fail) we'll ignore the cache?
- if any compiler emits any stderr, then now we'll warn on line 394? won't that happen all the time?

I _think_ this is what `okToCache` is supposed to encode: we shouldn't cache stuff that we don't want...but I am certainly missing something here.

Reply:
okToCache is only set on timeouts https://github.com/compiler-explorer/compiler-explorer/blob/5bbe1347f7b368f55c064115540910fda412c1ad/lib/exec.ts#L108

But you're right that it's currently noisy. But it's also quite informative. It informs about useless --help requests that will never result in a useful answer. For example in this last commit I removed all the useless --help calls that don't work with icc https://github.com/compiler-explorer/compiler-explorer/pull/5001/commits/85d0bf25180165d7a14fc69e6f6b659839b4e995

I will iterate over the warnings and eliminate where things are not needed and fix if needed. And will probably then remove the warning.

Reply:
removed it altogether, dont think we need it anymore